pkgs/development/python-modules/pysimplesoap/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, fetchpatch
+, fetchPypi
+, buildPythonPackage
+, m2crypto
+}:
+
+buildPythonPackage rec {
+  pname = "pysimplesoap";
+  # Unfortunately, the latest stable release is broken on Python 3.
+  version = "1.16.2";
+
+  src = fetchPypi {
+    pname = "PySimpleSOAP";
+    inherit version;
+    hash = "sha256-sbv00NCt/5tlIZfWGqG3ZzGtYYhJ4n0o/lyyUJFtZ+E=";
+  };
+
+  propagatedBuildInputs = [
+    m2crypto
+  ];
+
+  patches =
+    let
+      debianRevision = "5";  # The Debian package revision we get patches from
+      fetchDebianPatch = { name, hash }: fetchpatch {
```

Comment:
does it make sense to make this available to other packages?

Reply:
> Oh, does hydra automatically mirror things that were fetchurl'd or fetchpatch'd ?

Yes, it lands in the cache and the entire world pulls from there automatically.

But relying on cache as a permanent storage is considered bad-practice. Usually adding a webarchive url as a fallback is better than just cache, as cache just happens to be a permanent fod output storage.

Reply:
Seeing how complicated this is I'd leave it as out-of-scope for now, but if it makes sense you can create an issue/pr.